driver/result_set.cpp

```diff
@@ -67,8 +67,36 @@ void ColumnInfo::assignTypeInfo(const TypeAst & ast) {
     }
 }
 
-void ColumnInfo::updateTypeId() {
+void ColumnInfo::updateTypeInfo() {
     type_without_parameters_id = convertUnparametrizedTypeNameToTypeId(type_without_parameters);
+
+    auto tmp_type_name = convertTypeIdToUnparametrizedCanonicalTypeName(type_without_parameters_id);
+
+    if (
```

Comment:
Why not just add a 3 more lines to the `types_g` in `type_info.cpp`:?
```
{"Decimal", TypeInfo {"DECIMAL", false, SQL_DECIMAL, 1 + 2 + 38, 16}},
{"Decimal32", TypeInfo {"DECIMAL", false, SQL_DECIMAL, 1 + 2 + XXX, 4}},
{"Decimal64", TypeInfo {"DECIMAL", false, SQL_DECIMAL, 1 + 2 + YYY, 8}},
{"Decimal128", TypeInfo {"DECIMAL", false, SQL_DECIMAL, 1 + 2 + ZZZ, 16}},
```

Reply:
I guess, because they are just synonyms for people, and it is less error-prone to have as little duplication as possible in the metadata and keep it precise. Type handling code is terrible at the moment, should be reviewed when we allocate resources to that.